Sming/SmingCore/Platform/Station.cpp

```diff
@@ -241,6 +265,21 @@ String StationClass::getSSID()
 	return String((char*)config.ssid);
 }
 
+char * StationClass::getSSID(char * s, size_t bufSize)
+{
+	station_config config = {0};
+	if (!wifi_station_get_config(&config))
+	{
+		debugf("Can't read station configuration!");
+	}
+	else
+	{
+		debugf("SSID: %s", (char *)config.ssid);
+		strncpy(s, (char *)config.ssid, bufSize - 1);
```

Comment:
Because:

> If the end of the source C string (which is signaled by a null-character) is found before num characters have been copied, destination is padded with zeros until a total of num characters have been written to it.

and:

> No null-character is implicitly appended at the end of destination if source is longer than num.

I'm writing '\0' at the end of provided buffer in the case when `bufSize' is smaller than `sizeof(config.ssid)` and `config.ssid` would not fit entirely. Otherwise buffer will be padded with zeros anyway.